### DISCUSSION TITLE
请问这个是最终版本吗？

### DISCUSSION THREAD

**Discussion by leily127** [wu-kan/HPL-AI#3] [Q&A]
hello吴同学，
有幸拜读了您的毕业论文，想请问这份代码是论文中最终部署在鲲鹏+昇腾上最终版本的hpl-ai源码吗？

**Comment by wu-kan** [comment]
是的，在 #2 里也有其他人在昇腾上成功运行了，不同版本是通过切换不同宏定义来实现的，建议详细阅读下 [src/blas/HPLAI_blas.cc](https://github.com/wu-kan/HPL-AI/blob/master/src/blas/HPLAI_blas.cc) 源码。

**Comment by leily127** [comment]
好的好的 收到🫡谢谢您！！！🙏



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: WuK ***@***.***&gt;
发送时间: 2023年3月24日 15:18
收件人: wu-kan/HPL-AI ***@***.***&gt;
抄送: aleilei ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [wu-kan/HPL-AI] 请问这个是最终版本吗？ (Discussion #3)





 
是的，在 #2 里也有其他人在昇腾上成功运行了，不同版本是通过切换不同宏定义来实现的，建议详细阅读下 src/blas/HPLAI_blas.cc 源码。

**Comment by leily127** [comment]
嗨！吴坎同学！
不好意思打扰你了，能否请问一下你在编译blaspp库的时候有没有遇到类型转换出错的问题呀？我用spack安装出错后，自己去了bitbucket上用源码安装blaspp，就给我报错了。排查了一下问题都是出在类型转换这里🙏 如果你有时间的话回复我将十分感谢！！！



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: WuK ***@***.***&gt;
发送时间: 2023年3月24日 15:18
收件人: wu-kan/HPL-AI ***@***.***&gt;
抄送: aleilei ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [wu-kan/HPL-AI] 请问这个是最终版本吗？ (Discussion #3)





 
是的，在 #2 里也有其他人在昇腾上成功运行了，不同版本是通过切换不同宏定义来实现的，建议详细阅读下 src/blas/HPLAI_blas.cc 源码。

**Reply by wu-kan** [reply]
用 spack 装 blaspp 吧，bitbucket 上的源码是开发版，不一定是对的
